wisp/wisp.php

```diff
@@ -372,11 +394,74 @@ function wisp_CreateAccount(array $params) {
+                    $final_allocations = findFreePorts($available_allocations, $additional_port_list);
+
+                    if($final_allocations != false && $final_allocations['status'] == true) {
+                        $alloc_success = true;
+                        logModuleCall("WISP-WHMCS", "Successfully found an allocation. Setting primary allocation to ID " . $final_allocations['main_allocation_id'], "", "");
+                        
+                        $serverData['allocation']['default'] = intval($final_allocations['main_allocation_id']);
+                        $serverData['allocation']['additional'] = $final_allocations['additional_allocation_ids'];
+                        
+                        // Update the environment parameters - additional allocations
+                        foreach($final_allocations['additional_allocation_ports'] as $key => $port) {
+                            // If the key given in the config had a value of NONE, don't worry about adding it to the environment parameters.
+                            if(substr( $key, 0, 5 ) !== "NONE_") {
+                                $serverData['environment'][$key] = $port;
+                            }
+                        }
+                        // We successfully found and assigned an available allocation, break and check no more nodes.
+                        break;
+                    }
+                    logModuleCall("WISP-WHMCS", "Failed to find an available allocation on node: ".$node_id, "", "");
+                }
+                if(!$alloc_success) {
+                    // Failure handling logic
+                    if($additional_port_fail_mode = "Stop") {
+                        throw new Exception('Couldn\'t find any nodes to satisfy the requested allocations.');
+                    } else {
+                        // Continue with normal deployment
+                        $serverData['deploy']['port_range'] = $port_range;
+                    }
+                }
+            } else {
+                logModuleCall("WISP-WHMCS", "Unable to find any nodes at location ID: ".$loc_id, "", "");
+                throw new Exception('Couldn\'t find any nodes satisfying the request at location: ' . $loc_id);
+            }
+        } else {
+            // Continue with normal deployment
+            $serverData['deploy']['port_range'] = $port_range;
+        }
 
+        // Create the game server
+        $server = wisp_API($params, 'servers', $serverData, 'POST');
         if($server['status_code'] === 400) throw new Exception('Couldn\'t find any nodes satisfying the request.');
         if($server['status_code'] !== 201) throw new Exception('Failed to create the server, received the error code: ' . $server['status_code'] . '. Enable module debug log for more info.');
 
+        // Catch API errors
```

Comment:
This code block is already 2 lines above?

Reply:
Thank you, fixed 👌